### PR TITLE
Add system metrics data for export

### PR DESCRIPTION
**Pull request by rolfyone**
partially addresses #4391

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java [hunk 1]
```diff
@@ -13,57 +13,220 @@
...
+
+  @JsonProperty("misc_os")
+  private final String miscOs = getNormalizedOSVersion();
+
+  public SystemMetricData(long timestamp, final File beaconNodeDirectory) {
+    super(timestamp, SYSTEM.getDisplayName());
+
+    SystemInfo systemInfo = new SystemInfo();
+    final HardwareAbstractionLayer hardware = systemInfo.getHardware();
+    hardware.getProcessor().getPhysicalProcessorCount();
```
Comment (ajsutton) [inline data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java]: This line doesn't seem to be doing much.

File: data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java [hunk 2]
```diff
@@ -13,57 +13,220 @@
...
+      case "win":
+        return new WindowsOperatingSystem().getSystemBootTime();
+      case "mac":
+        return new MacOperatingSystem().getSystemBootTime();
+      default:
+        return 0L;
+    }
+  }
+
+  private long[] setProcessLoadTicks(final long[][] processorCpuLoadTicks) {
```
Comment (ajsutton) [inline data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java]: This doesn't appear to set anything.  Maybe should be `calculateProcessLoadTicks`?

File: data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java [hunk 3]
```diff
@@ -13,57 +13,220 @@
...
+    iowait /= 1000;
+    // note: currently beaconcha.in expects system time to be everything
+    return new long[] {user, (system + user + idle + iowait), idle, iowait};
+  }
+
+  private static String getNormalizedOSVersion() {
+    String currentVersionInfo = VersionProvider.VERSION;
+    if (currentVersionInfo.contains("linux")) return "lin";
+    if (currentVersionInfo.contains("windows")) return "win";
+    if (currentVersionInfo.contains("osx")) return "mac";
```
Comment (ajsutton) [inline data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java]: I know this is nice and short but we always use braces on if statements.  Probably winds up working best as a series of if/else if

**Review by ajsutton — APPROVED** [review]
LGTM. Just a few nits.

File: data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java (hunk 2 above)
Comment (rolfyone) [reply data/publisher/src/main/java/tech/pegasys/teku/data/publisher/SystemMetricData.java]: fair point, name is a victim of refactoring.. will change.

**Pull request merged**